v2/blockstore/readwrite.go

```diff
@@ -57,7 +57,12 @@ func WithIndexPadding(p uint64) Option {
 	}
 }
 
+// TODO When we are constructing a UnxFS DAG from a user file using this read-write blockstore as the staging store,
+// we might not have the root Cid will we finalise.
+// Please can we add an option to skip the root here and give it when we finalize the DAG ?
 // NewReadWrite creates a new ReadWrite at the given path with a provided set of root CIDs as the car roots.
+//
+// TODO We need an option/feature here to avoid writing/Putting duplicates.
```

Comment:
We'll make this the default behavior, to keep the set of blocks and the index in sync.

Reply:
I've started a thread on this in Slack, and will open an issue afterwards.